addons/website_sale/views/templates.xml

```diff
@@ -1994,4 +1994,18 @@
             <t t-set="html_data" t-value="dict(html_data, **{'data-add2cart-redirect': website.cart_add_on_page and '1' or '0'})"/>
         </xpath>
     </template>
+
+    <template id="product_category_extra_link" name="Product Category Extra Link">
+        <t t-if="len(categories) == 1">
+            <button class="btn btn-link btn-sm" t-out="title + ' ' + categories.name"
+                    t-attf-onclick="location.href='/shop/category/#{slug(categories)}';return false;"/>
+        </t>
+        <t t-else="len(categories) > 1">
+            <button class="btn btn-link btn-sm pr-0" disabled="disabled" t-out="title"/>
+            <t t-foreach="categories" t-as="category">
+                <button class="btn btn-link btn-sm p-0" t-out="category.name"
+                    t-attf-onclick="location.href='/shop/category/#{slug(category)}';return false;"/>
+            </t>
+        </t>
+    </template>
```

Comment:
Feel free to unify both. "Category:" being within the text was not a requirement per say - it was just the behavior or the `extra_link` + `extra_link_url` which does not allow for more detail.
This being an additional field, I think you can always have the title that depends on the length, then always have the loop for all categories, which might be a single one. This should also solve your slightly different layout issue.

Reply:
Thinking now about whether this is OK to include in a fix, I believe it is:
it does not impact the old template xpaths - even though the old `extra_link` will not be populated anymore for products

Reply:
I fear we fail to respect the stable policy with this one.

Indeed the python code will depend on the existence of the template which should be avoided AFAIK.
Maybe this fix should target master and we have to find another way to handle this in stable ?

Reply:
Done in #79363